code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,712 @@
+ 
+security:
+  - oAuth2ClientCredentials: 
+    - nbi-mgmt
+ 
+tags:
+  - name: Application
+    description: Application Management
+  - name: Application Instance 
+    description: Application Instance Management
+  - name: Edge Cloud
+    description: Edge Cloud Nodes Available
+ 
+paths:
+  /appInstance:
+    post: 
+      tags:
+        - Application Instance
+      summary: Instantiation of an Application
+      description: Ask the OP to instantiate an application to one or several Edge Cloud Nodes with an Application as an input and an Application Instance as the output. 
+      operationId: appInstantiation
+      requestBody:
+        description: Details regarding where the application should be instantiated
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AppInstantiation'
+        required: true      
+      responses:
+        '201':
+          description: Application instantiated successfully
+          content: 
+            application/json:
+              schema:
+                type: object
+                properties:
+                  appId:
+                    $ref: '#/components/schemas/AppId'
+                  aplicationInstace:
+                    type: array
+                    items:
+                      $ref:  '#/components/schemas/InstantiatedApp'
+                minItems: 1                 
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '409':
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorInfo'
+              example:
+                status: 409
+                code: CONFLICT
+                message: "Application already instantiated in the given Edge or Region"
+        '500':
+          $ref: '#/components/responses/500'
+        '501':
+          $ref: '#/components/responses/501'
+        '503':
+          $ref: '#/components/responses/503'
+  /appInstance/{appId}:
+    get:
+      tags:
+        - Application Instance
+      summary: Retrieve the information of Application Instances for a given App
+      description: Ask the OP the information of the instances for a given application
+      operationId: getAppInstance
+      parameters:
+        - name: appId
+          description: A globally unique identifier associated with the application. OP generates this identifier when the application is submitted over NBI.
+          in: path
+          required: true
+          schema: 
+              $ref: '#/components/schemas/AppId'
+        - name: appInstanceId
+          description:  A globally unique identifier associated with a running instance of an application. OP generates this identifier. 
+          in: query
+          required: false
```

Comment:
Do we need to make "required" as true? Otherwise we need to define if no instance id has been provided then what would be the behavior?  To terminate all the instances of this app?

Reply:
The intention is that all the instances can be deleted if the instanceID is no specificated. This behaviour could be explained in the documentation